### ISSUE TITLE
[Support]: Transcoding historical recording files from HEVC to H264

### ISSUE THREAD

**Issue by PeteBa** [blakeblackshear/frigate#3349]
### Describe the problem you are having

My cameras only support HEVC so I am unable to playback recordings from within the Frigate GUI. The recordings will playback using VLC and show the correct HEVC codec information.

To work-around this problem, I have used ffmpeg to transcode historic HEVC recordings and replace them with H264 versions. These playback correctly in VLC and show the expected H264 codec information. For the most part, this also allows recordings to playback from within the Frigate GUI which is great.

However, with the H264 files, there are some situations where Frigate playback skips the first 10-second recording and only plays back from the second 10-second file onward. I suspect this has something to do with timestamps and that Frigate cant locate the start of the event in the first segment and so skips forward to the second segment. This is just my speculation but perhaps the transcoding is inadvertently dropping some information that Frigate relies upon.

I am using the following command to do the transcoding and then replacing the original with the new recording:
```
ffmpeg -i 24.12.mp4 -c:v h264 new/24:12.mp4
```

Any thoughts as to what might be going wrong ?

### Version

0.10.1-83481AF

### Frigate config file

```yaml
detectors:
  coral:
    type: edgetpu
    device: pci

ffmpeg:
  hwaccel_args: -hwaccel vaapi -hwaccel_device /dev/dri/renderD128 -hwaccel_output_format yuv420p

cameras:
  deckcam:
    ffmpeg:
      inputs:
        - path: rtsp://deckcam:554/Streaming/Channels/102?transportmode=unicast&profile=Profile_2
          roles:
            - detect
        - path: rtsp://@deckcam:554/Streaming/Channels/101?transportmode=unicast&profile=Profile_1
            - record
    detect:
      width: 1280
      height: 720
      fps: 4
    rtmp:
      enabled: False
    motion:
      mask:
        - 0,565,0,0,394,0,372,45,188,128,228,452,21,601
        - 1280,720,1251,720,1280,720,1280,0,579,0,604,36,750,35,714,124,702,513,1164,544
    record:
      enabled: True
    snapshots:
      enabled: True
      timestamp: True
      bounding_box: True
```

### Relevant log output

```shell
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[2022-06-15 13:55:11] frigate.app                    INFO    : Starting Frigate (0.10.1-83481af)
Starting migrations
[2022-06-15 13:55:11] peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
[2022-06-15 13:55:12] peewee_migrate                 INFO    : There is nothing to migrate
[2022-06-15 13:55:12] detector.coral                 INFO    : Starting detection process: 218
[2022-06-15 13:55:12] frigate.app                    INFO    : Output process started: 220
[2022-06-15 13:55:12] ws4py                          INFO    : Using epoll
[2022-06-15 13:55:12] frigate.edgetpu                INFO    : Attempting to load TPU as pci
[2022-06-15 13:55:12] frigate.app                    INFO    : Camera processor started for deckcam: 224
[2022-06-15 13:55:12] frigate.app                    INFO    : Camera processor started for gardencam: 226
[2022-06-15 13:55:12] frigate.app                    INFO    : Camera processor started for alleycam: 227
[2022-06-15 13:55:12] frigate.app                    INFO    : Camera processor started for frontcam: 231
[2022-06-15 13:55:12] frigate.app                    INFO    : Capture process started for deckcam: 243
[2022-06-15 13:55:12] frigate.edgetpu                INFO    : TPU found
[2022-06-15 13:55:12] frigate.app                    INFO    : Capture process started for gardencam: 247
[2022-06-15 13:55:12] frigate.app                    INFO    : Capture process started for alleycam: 252
[2022-06-15 13:55:12] frigate.app                    INFO    : Capture process started for frontcam: 253
[2022-06-15 13:55:12] ws4py                          INFO    : Using epoll
[2022-06-15 13:55:18] frigate.record                 WARNING : Discarding a corrupt recording segment: {'cache_path': '/tmp/cache/gardencam-20220615135412.mp4', 'start_time': datetime.datetime(2022, 6, 15, 13, 54, 12)}
[2022-06-15 13:55:18] frigate.record                 WARNING : Discarding a corrupt recording segment: {'cache_path': '/tmp/cache/gardencam-20220615135412.mp4', 'start_time': datetime.datetime(2022, 6, 15, 13, 54, 12)}
[2022-06-15 13:55:18] frigate.record                 WARNING : Discarding a corrupt recording segment: {'cache_path': '/tmp/cache/gardencam-20220615135412.mp4', 'start_time': datetime.datetime(2022, 6, 15, 13, 54, 12)}
[2022-06-15 13:55:18] frigate.record                 WARNING : Discarding a corrupt recording segment: {'cache_path': '/tmp/cache/gardencam-20220615135412.mp4', 'start_time': datetime.datetime(2022, 6, 15, 13, 54, 12)}

```

### FFprobe output from your camera

```shell
ORIGINAL RECORDING

ffprobe orig/24.12.mp4
ffprobe version 4.3.1 Copyright (c) 2007-2020 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.3.0-17ubuntu1~20.04)
  configuration: --disable-debug --disable-doc --disable-ffplay --enable-shared --enable-avresample --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-gpl --enable-libfreetype --enable-libvidstab --enable-libmfx --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libxcb --enable-libx265 --enable-libxvid --enable-libx264 --enable-nonfree --enable-openssl --enable-libfdk_aac --enable-postproc --enable-small --enable-version3 --enable-libzmq --extra-libs=-ldl --prefix=/opt/ffmpeg --enable-libopenjpeg --enable-libkvazaar --enable-libaom --extra-libs=-lpthread --enable-vaapi --extra-cflags=-I/opt/ffmpeg/include --extra-ldflags=-L/opt/ffmpeg/lib
  libavutil      56. 51.100 / 56. 51.100
  libavcodec     58. 91.100 / 58. 91.100
  libavformat    58. 45.100 / 58. 45.100
  libavdevice    58. 10.100 / 58. 10.100
  libavfilter     7. 85.100 /  7. 85.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  7.100 /  5.  7.100
  libswresample   3.  7.100 /  3.  7.100
  libpostproc    55.  7.100 / 55.  7.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'orig/24.12.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2mp41
    title           : Media Presentation
    encoder         : Lavf58.45.100
  Duration: 00:00:09.88, start: 0.000000, bitrate: 3594 kb/s
    Stream #0:0(und): Video: hevc (hev1 / 0x31766568), yuvj420p(pc), 1920x1080, 3592 kb/s, 8.09 fps, 90k tbr, 90k tbn, 90k tbc (default)
    Metadata:
      handler_name    : VideoHandler

TRANSCODED RECORDING

# ffprobe new/24.12.mp4
ffprobe version 4.3.1 Copyright (c) 2007-2020 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.3.0-17ubuntu1~20.04)
  configuration: --disable-debug --disable-doc --disable-ffplay --enable-shared --enable-avresample --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-gpl --enable-libfreetype --enable-libvidstab --enable-libmfx --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libxcb --enable-libx265 --enable-libxvid --enable-libx264 --enable-nonfree --enable-openssl --enable-libfdk_aac --enable-postproc --enable-small --enable-version3 --enable-libzmq --extra-libs=-ldl --prefix=/opt/ffmpeg --enable-libopenjpeg --enable-libkvazaar --enable-libaom --extra-libs=-lpthread --enable-vaapi --extra-cflags=-I/opt/ffmpeg/include --extra-ldflags=-L/opt/ffmpeg/lib
  libavutil      56. 51.100 / 56. 51.100
  libavcodec     58. 91.100 / 58. 91.100
  libavformat    58. 45.100 / 58. 45.100
  libavdevice    58. 10.100 / 58. 10.100
  libavfilter     7. 85.100 /  7. 85.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  7.100 /  5.  7.100
  libswresample   3.  7.100 /  3.  7.100
  libpostproc    55.  7.100 / 55.  7.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '24.12.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2avc1mp41
    title           : Media Presentation
    encoder         : Lavf58.45.100
  Duration: 00:00:10.13, start: 0.000000, bitrate: 3234 kb/s
    Stream #0:0(und): Video: h264 (avc1 / 0x31637661), yuvj420p(pc), 1920x1080, 3233 kb/s, 8.09 fps, 8.09 tbr, 7200k tbn, 16.19 tbc (default)
    Metadata:
      handler_name    : VideoHandler
```

### Operating system

UNRAID

### Install method

Docker CLI

### Coral version

M.2

### Network connection

Wired

### Camera make and model

Annke

**Comment by NickM-27** [comment]
Providing logs would be helpful here to see what is going on.

Just to make sure there is understanding, when you check the recording segments manually there is one inside the `recordings` directory that is not being played when playing back events?

**Comment by PeteBa** [comment]
Yes, all the recording segments are present.

For example, for my frontcam between 11:00-12:00 today, I have five transcoded segments each containing 10 seconds of footage in h264 format showing continuity of embedded timestamps during the period and visually identical to the HEVC recorded by Frigate. Confirmed using vlc:
```
root@/media/frigate/recordings/2022-06/15/11/frontcam# ls
24.12.mp4  24.22.mp4  24.32.mp4  57.12.mp4  57.22.mp4
```
Frigate GUI shows two events for that hour:

1) Person event from 11:24:21 to 11:24:45 - playback seems to work correctly showing footage with embedded timestamps from 11:24:20 to11:20:40. Note the camera clock is about 3-4 seconds slow compared with the server. So I would have expected to see a couple of seconds of earlier footage but that could be because recordings need to start on an i-frame (?). The event stop and embedded timestamps correlate given the slow clock.

2) Person event from 11:57:12 to 11:57:30 - this incorrectly skips all but the last three seconds of the first segment with embedded timestamps from 11:57:17-20. It then "plays" a frozen frame (last frame of segment one) for eight seconds before going on to play the second segment showing timestamps from 11:57:21-11:57:28. Again the stop time correlates given camera clock.

In the second example, It looks like it is indexing into the first segment at the wrong place and then padding with the last frame until it goes onto the second segment. Could this be something to do with framerate? FFProbe says frame rate is 8fps in both the HEVC and H264 files as set in my camera.

I have added a log file that covers playback of these events but nothing is logged. I increased logging to debug level but didnt see anything related to the playback engine. Nothing showed up in Chrome developer tools either. Is there a way to get diagnostics from the playback component as I think that is browser based right ?
